src/components/Register.js

```diff
@@ -1,58 +1,158 @@
 import React, { useState, useEffect } from 'react';
 import axios from 'axios';
+import {useHistory} from 'react-router-dom'
 import {InputGroup, Form, FormControl, Button, Alert} from 'react-bootstrap';
+import Modal from 'react-bootstrap/Modal'
+import {register, getUser} from "../api"
 
-import {fetchUser, getToken, setToken as setLocalToken} from '../util'
+import {getLocalToken, setLocalToken} from '../util'
 
 
 export default props => {
-  const {setUser, setToken, token} = props;
+  const {user, setUser, token, setToken} = props;
+  const history = useHistory()
   const [username, setUsername] = useState('');
   const [password, setPassword] = useState('');
-  const [error, setError] = useState();
-    
-  useEffect(() => {
-    if (!token) {
-      setToken(getToken())
-    } 
-  },[getToken]) 
+  const [ email, setEmail ] = useState('')
+  const [ firstName, setFirstName] = useState('');
+  const [ lastName, setLastName] = useState('')
+  const [message, setMessage] = useState();
+  const [show, setShow] = useState(true);
+  const handleClose = () => setShow(false);
```

Comment:
DRY programming! 
Another good opportunity  for a reusable component is for a `modal`. Make one, and reuse it everywhere!